pytorch_lightning/loggers/comet.py

```diff
@@ -240,11 +240,11 @@ def log_hyperparams(self, params: Union[Dict[str, Any], Namespace]) -> None:
     def log_metrics(self, metrics: Dict[str, Union[torch.Tensor, float]], step: Optional[int] = None) -> None:
         assert rank_zero_only.rank == 0, "experiment tried to log from global_rank != 0"
         # Comet.ml expects metrics to be a dictionary of detached tensors on CPU
-        for key, val in metrics.items():
+        metrics_without_epoch = metrics.copy()
+        for key, val in metrics_without_epoch.items():
             if is_tensor(val):
-                metrics[key] = val.cpu().detach()
+                metrics_without_epoch[key] = val.cpu().detach()
 
-        metrics_without_epoch = metrics.copy()
         epoch = metrics_without_epoch.pop("epoch", None)
```

Comment:
The tensor conversion might not be necessary, at least unless this function is called manually.

The logger connector converts all tensors  to scalars before calling this function

https://github.com/PyTorchLightning/pytorch-lightning/blob/f79993a705e4d2f187739751c00a677aedbfabfd/pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py#L94-L95

Reply:
Oh, so should or should I not do the scalar conversion? Should I keep the tensor conversion in case the function is called manually?

Reply:
> Should I keep the tensor conversion in case the function is called manually?

Any thoughts on this @awaelchli ?

Regardless of whether we convert again or not, you could use `metrics_to_scalars` to do it

Reply:
Oh alright! 

Also if you do not mind, can you have a look at the test I added? It is failing, I looked at the logs, and I am having  difficulty in determining what could I do to resolve the error. 

Furthermore,  I tried running the new test locally on my machine, however, I am getting an error:
```
/opt/python/3.8.6/lib/python3.8/bz2.py:19: in <module>
    from _bz2 import BZ2Compressor, BZ2Decompressor
E   ModuleNotFoundError: No module named '_bz2'
```
I searched online and tried quite a few things in vain, however have still not been able to resolve this error. My python version is 3.8.6.

Reply:
Hi @carmocca 
I looked at the [`metrics_to_scalars`](https://github.com/PyTorchLightning/pytorch-lightning/blob/f79993a705e4d2f187739751c00a677aedbfabfd/pytorch_lightning/utilities/metrics.py#L24
) function definition here and it seems that it can only be used for converting single element tensors into scalars. 

So will the tensors being passed as input to the `logger.log_metric()` always be single element tensors? If not, then I don't think that function might be useful in that scenario.

Reply:
Hey, see my [original comment](https://github.com/PyTorchLightning/pytorch-lightning/issues/7021#issuecomment-821981146) from the issue. What this PR here is fixing (and what the issue reported) is a direct call to `logger.log_metrics`, so we don't need to think about any other Lightning internal call here.

Reply:
I see, so to confirm, I should not do any more changes to the `log_metrics` function, right? Since there was a direct call to `logger.log_metrics`.

Reply:
Yes, the fix you added is good like this in my opinoin. Plus, the test you added fails without the fix applied, and that's what we wanted to fix.